doc/tensor.md

```diff
@@ -1327,36 +1331,66 @@ x:indexFill(2,torch.LongTensor{4,2},-10)
 ```
 
 <a name="torch.Tensor.gather"/>
-### [Tensor] gather(src, dim, index) ###
+### [Tensor] gather(dim, index) ###
```

Comment:
It's actually quite common to be able to call a function both as `src:func()` and as `dst:func(src)`. However, often the first case means essentially `src:func(src)`, i.e. it's an in-place operation. This is where `gather` differs.

I'd say the most common uses of `gather` will be `dst:gather(src, dim, idx)`, because you want to avoid memory allocations where possible. So maybe that's an argument for leaving `src` in the header :)